lib/rubocop/cop/style/nested_interpolation.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks nested interpolations
+      #
+      # @example
+      #
+      #   # bad
+      #   "Hello, #{user.blank? ? 'guest' : "dear #{user.name}"}"
+      #
+      #   # good
+      #   user_name = user.blank? ? 'guest' : "dear #{user.name}"
+      #   "Hello, #{user_name}"
+      class NestedInterpolation < Cop
```

Comment:
I guess this should named `RedundantNestedInterpolation` to be aligned with the typical naming scheme we employ.